lib/spire/order.rb

```diff
@@ -1,94 +1,218 @@
 module Spire
-  # A Customer is a customer account.
+  # An Order is a record that belongs to a Customer
   #
   # @!attribute [r] id
   #   @return [int]
-  # @!attribute [rw] background_color As a color decimal
-  #   @return [int]
-
+  # @!attribute [r] order_no
+  #   @return [String]
   # @!attribute [rw] customer
   #   @return [Hash]
-  # @!attribute [rw] address 
+  # @!attribute [rw] status
+  #   @return [string]
+  # @!attribute [rw] type
+  #   @return [String]
+  # @!attribute [rw] hold
+  #   @return [Boolean]
+  # @!attribute [rw] order_date
+  #   @return [String]
+  # @!attribute [rw] address
   #   @return [Hash]
   # @!attribute [rw] shipping_address
   #   @return [Hash]
+  # @!attribute [rw] customer_po
+  #   @return [String]
+  # @!attribute [rw] fob
+  #   @return [String]
+  # @!attribute [rw] terms_code
+  #   @return [String]
+  # @!attribute [rw] terms_text
+  #   @return [String]
+  # @!attribute [rw] freight
+  #   @return [String]
+  # @!attribute [rw] taxes
+  #   @return [Array]
+  # @!attribute [rw] subtotal
+  #   @return [String]
+  # @!attribute [rw] subtotal_ordered
+  #   @return [String]
+  # @!attribute [rw] discount
+  #   @return [String]
+  # @!attribute [rw] total_discount
+  #   @return [String]
+  # @!attribute [rw] total
+  #   @return [String]
+  # @!attribute [rw] total_ordered
+  #   @return [String]
+  # @!attribute [rw] gross_profit
+  #   @return [String]
   # @!attribute [rw] items
   #   @return [Array]
+  # @!attribute [r] created
+  #   @return [String]
+  # @!attribute [r] modified
+  #   @return [String]
+  # @!attribute [rw] background_color
+  #   @return [String]
+  # @!attribute [r] created_by
+  #   @return [String]
+  # @!attribute [r] modified_by
+  #   @return [String]
 
   class Order < BasicData
-    register_attributes :id, :background_color, :customer, :address, :shippingAddress, :items, :created_by, :modified_by,
+    register_attributes :id, :order_no, :customer, :status, :type, :hold,
+      :order_date, :address, :shipping_address, :customer_po, :fob, :terms_code,
+      :terms_text, :freight, :taxes, :subtotal, :subtotal_ordered, :discount,
+      :total_discount, :total, :total_ordered, :gross_profit, :items, :created_by,
+      :modified_by, :created, :modified, :background_color,
       readonly: [
-        :id
+        :created_by, :modified_by, :created, :modified, :order_no
       ]
 
-    validates_presence_of :customer, :items
+    validates_acceptance_of :id, :customer, :address, :shipping_address, :items
+
+    ACTIVE = "0"
+    ON_HOLD = "1"
+    INACTIVE = "2"
 
     SYMBOL_TO_STRING = {
       id: 'id',
-      background_color: 'backgroundColor', 
+      order_no: 'orderNo',
       customer: 'customer',
+      status: 'status',
+      type: 'type',
+      hold: 'hold',
+      order_date: 'orderDate',
       address: 'address',
-      shipping_address: "shippingAddress",
+      shipping_address: 'shippingAddress',
+      customer_po: 'shippingPO',
```

Comment:
@Dujota I feel like these should be called the same. Whatever the API uses, whether it be

```shipping_po: 'shippingPO'```

or

```customer_po: 'customerPO'```

Reply:
small typo, should be customerPO